flink-python/src/main/java/org/apache/flink/python/AbstractPythonFunctionRunner.java

```diff
@@ -176,10 +179,29 @@ public void open() throws Exception {
 		Struct pipelineOptions = PipelineOptionsTranslation.toProto(portableOptions);
 
 		jobBundleFactory = createJobBundleFactory(pipelineOptions);
-		stageBundleFactory = jobBundleFactory.forStage(createExecutableStage());
+		stageBundleFactory = createStageBundleFactory();
 		progressHandler = BundleProgressHandler.ignored();
 	}
 
+	/**
+	 * To make the error messages more user friendly, read the boot logs from the temp log file and
+	 * throw an exception directly.
+	 */
+	private StageBundleFactory createStageBundleFactory() throws Exception {
+		try {
+			return jobBundleFactory.forStage(createExecutableStage());
+		} catch (Exception e) {
+			byte[] output =
+				Files.readAllBytes(
+					new File(((ProcessPythonEnvironmentManager) environmentManager).getBaseDirectory()
```

Comment:
What happens if it's not an instance of ProcessPythonEnvironmentManager?

Reply:
How about adding the `getBaseDirectory` method into the interface?